src/playlist.js

```diff
@@ -0,0 +1,66 @@
+var playItem = require('./playitem.js');
+var setupAutoadvance = require('./autoadvance.js');
+
+var playlistMaker = function(player, plist) {
+  var currentIndex = 0;
+  var autoadvanceTimeout = null;
+
+  var playlist = function playlist(list) {
+    if (typeof list === 'number') {
+      return player.playlist.item(list);
+    }
+
+    this.playlist = playlistMaker(this, list);
+  };
+
+  playlist.list = function list() {
+    return plist.slice();
+  };
+
+  playlist.item = function item(index) {
+    if (typeof index === 'number' && index >= 0 && index < plist.length) {
+      currentIndex = index;
+      return playItem(player, autoadvanceTimeout, plist[currentIndex]);
```

Comment:
Nope. I think this is still partly in POC stage. There are no tests or anything, though, that's the very next thing I'll be working on. As part of that, we'd want to solidify the API.
Assuming that everything looks mostly ok, I'd like to merge this branch before starting on the tests and API improvements in another branch.